mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxView.kt

```diff
@@ -29,10 +32,12 @@ interface MvRxView : LifecycleOwner {
      * a lifecycle event (e.g. rotation) you should assign a consistent id. Likely this means you should save the id
      * in onSaveInstance state. The viewId will not be accessed until a subscribe method is called.
      * Accessing mvrxViewId before calling super.onCreate() will cause a crash.
-     *
-     * For an example implementation see [BaseMvRxFragment].
      */
-    val mvrxViewId: String
+    val mvrxViewId: String get() = when (this) {
+        is ViewModelStoreOwner -> ViewModelProvider(this).get(MvRxViewIdViewModel::class.java).mvrxViewId
```

Comment:
This is the key bit of the PR. It stores the view id in a ViewModel with a SavedStateHandle to maintain the id across process restarts.

Reply:
@gpeal i guess we're not worried about the extra overhead this adds to lookup the id every time we need to reference it?

Reply:
@elihart It should just be a map lookup so I think it's okay.